pkg/controllers/dynakube/apimonitoring/reconciler.go

```diff
@@ -45,17 +47,33 @@ func (r *Reconciler) createObjectIdIfNotExists(ctx context.Context) (string, err
 		return "", errors.New("no kube-system namespace UUID given")
 	}
 
-	monitoredEntitiesreconciler := monitoredentities.NewReconciler(r.dtc, r.dk)
-
-	err := monitoredEntitiesreconciler.Reconcile(ctx)
+	err := r.monitoredEntitiesReconciler(r.dtc, r.dk).Reconcile(ctx)
 	if err != nil {
 		return "", err
 	}
 
-	monitoredEntity := []dtclient.MonitoredEntity{
-		{
-			EntityId: r.dk.Status.KubernetesClusterMEID,
-		},
+	var monitoredEntity []dtclient.MonitoredEntity
+
+	if r.dk.Status.KubernetesClusterMEID != "" {
+		monitoredEntity = []dtclient.MonitoredEntity{
+			{
+				EntityId: r.dk.Status.KubernetesClusterMEID,
+			},
+		}
+	}
+
+	objectID, err := r.dtc.CreateOrUpdateKubernetesSetting(ctx, r.clusterLabel, r.dk.Status.KubeSystemUUID, r.dk.Status.KubernetesClusterMEID)
```

Comment:
can you move this to the else branch of the above if? because if you have MEID you already have KubernetesSetting Object correct?

Reply:
If i get you correctly: The function would only be called if MEID is not set -> create part. If it is set we should not call the function: What about the update part then? Am i missing something here?